lib/new_relixir/agent.ex

```diff
@@ -138,7 +138,7 @@ defmodule NewRelixir.Agent do
     Enum.join(segments)
   end
 
-  defp urljoin([]), do: []
+  defp urljoin(list) when is_list(list) and length(list) == 0, do: []
```

Comment:
Fixes
```elixir
lib/new_relixir/agent.ex:141: 
The pattern [] can never match the type [any(),...]
```

Reply:
The actual length of the list it is not important, so you can use list == [] to avoid hitting each item. But in this case, I think this is not the problem itself. We need to check the cause why it is happening.

Reply:
@rafaelrochasilva  Oh, I see! Dialyzer checks that this function is called over a list that is never empty. As you can see [here](https://github.com/edipox/new-relixir/blob/9a0635fa337b355a249bd0f337497a291339a117/lib/new_relixir/agent.ex#L137)
we call it with 
```elixir
args ++ base_args
```
which results in a non-empty list given the value of [`base_args`](https://github.com/edipox/new-relixir/blob/9a0635fa337b355a249bd0f337497a291339a117/lib/new_relixir/agent.ex#L131)